ui/src/sources/components/DashboardStep.tsx

```diff
@@ -196,20 +240,33 @@ class DashboardStep extends Component<Props, State> {
     const {source} = this.props
 
     if (source) {
-      const suggestedProtoboardsList = await getSuggestedProtoboards(
-        source,
-        protoboards
-      )
-
-      if (suggestedProtoboardsList.length === 0) {
-        return
+      if (this.isComponentMounted) {
```

Comment:
I don't see how we can get to this line if component is not mounted?

Reply:
it has to be with the component becoming unounted during the async call.